### DISCUSSION TITLE
What are the platforms?

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre-native#358] [Q&A]
MapLibre GL Native has multiple platform bindings. What are they for and how are they related to eachother?

The `platform/` folder looks roughly like this:
```
android
darwin
default
glfw
ios
  platform
    darwin
    default
    ios
    macos
linux
macos
node
qt
```

**Comment by wipfli** [comment]
Some stuff from the different READMEs:

* `android`: The MapLibre Maps SDK for Android is a library based on MapLibre GL Native for embedding interactive map views with scalable, customizable vector maps onto Android devices.
* `darwind`: The code in the Darwin platform targets Apple platforms but is not specific to iOS or macOS. This code is not distributed as an SDK in itself, but is required by iOS and macOS builds of MapLibre GL Native and ultimately by the [MapLibre Maps SDK for iOS](https://github.com/maplibre/maplibre-gl-native/tree/main/platform/ios) and [MapLibre Maps SDK for macOS](https://github.com/maplibre/maplibre-gl-native/tree/main/platform/macos).
* `default`: 
* `glfw`:
* `ios`: The MapLibre Maps SDK for iOS is an open-source framework for embedding interactive map views with scalable, customizable vector maps into Cocoa Touch applications on iOS 9.0 and above using Objective-C, Swift, or Interface Builder.
* `linux`: A simple executable and test suite for Linux based on [MapLibre GL Native](https://github.com/maplibre/maplibre-gl-native/blob/main/README.md).
* `macos`: This folder contains build configuration files for the macOS distribution of MapLibre GL Native, which ultimately targets Cocoa applications written in Objective-C, Swift, Interface Builder, or AppleScript via the MapLibre Maps SDK for macOS.
* `node`:
* `qt`: This is a community maintained MapLibre SDK for usage in Qt apps.

**Reply by hy9be** [reply]
``glfw``is a library for creating local windows, and receiving input and events for OpenGL. I am not sure about the original intention but it looks like related to lightweight local debugging.

**Reply by hy9be** [reply]
I plan to update the README and reorganize documents a bit like what I did for the JS repo. Will include this somewhere if this is not covered in any existing markdown docs.

**Reply by ueoeae** [reply]
Are there plans to support building on windows without QT?

**Reply by ntadej** [reply]
We need core platform support that is not based on Node or Qt. Currently we have no expert to implement that.

**Reply by pierr3** [reply]
I'm a tad confused about this - shouldn't the glfw platform be that universal core platform support? I want to use maplibre native in headless mode across desktop platforms to generate a texture I'll then display myself with OpenGL, but cannot find how to build / whether the glfw platform would be re-hashable for this.

**Reply by louwers** [reply]
We actually have Windows support now @pierr3 

https://github.com/maplibre/maplibre-native/blob/main/platform/glfw/CMakeLists.txt#L64

Give it a spin and feel free to report back.

**Comment by pierr3** [comment]
Oh that’s great, thank you - i’ll give it a go.

Any pointers as to what would be needed to start supporting macOS?

**Reply by louwers** [reply]
Have a look at this: https://gist.github.com/gcatlin/987be74e2d58da96093a7598f3fbfb27

**Reply by louwers** [reply]
You can reply in the same thread. :)

When we bring back support for macOS (& AppKit) we will probably use the Bazel (instead of CMake) for building, this is also what we use for iOS right now.

**Reply by pierr3** [reply]
Apologies I was replying through email directly and turns out GH doesn't put that reply in the same thread haha

**Reply by louwers** [reply]
Ah now it makes sense. 😄

**Comment by pierr3** [comment]
Thank you, I’m more concerned about finding a cmake build path on macos for
maplibre-native, as I want to have c/c++ bindings to avoid duplicating my
code with a lot of objective-c. I currently see that the build path is
through xcode for obj-c bindings, but I have not dug deep enough.

**Reply by acalcutt** [reply]
I think glfw build used to be part of platform/macos/macos.cmake , which has been removed in main with the metal changes

Similar to my issue with the node version and macos. I was trying to see if I could get it working again for node but didn't really have the c++ knowledge
https://github.com/maplibre/maplibre-native/pull/1793/files#diff-ceabd8d690fb4d16fcdd7e01b6d9865ba826c85a4bbdce00be67e69a3c34a501

**Reply by acalcutt** [reply]
The opengl-2 branch still should have macos support, glfw may work there on macos with cmake.
https://github.com/maplibre/maplibre-native/tree/opengl-2
